### PR TITLE
MQTT streaming: put routers under connector actors

### PR DESCRIPTION
**Pull request by ennru**
## Purpose

Move the router actors to be children of the `ClientState` and `ServerState` instead of system actors as part of reducing the Akka classic vs Akka typed mixup

## Background Context

As #2097 shows, the fact that Akka Typed is not binary-compatible between Akka 2.5 and 2.6 bites us.

### CODE REVIEW

File: mqtt-streaming/src/main/scala/akka/stream/alpakka/mqtt/streaming/scaladsl/MqttSession.scala [hunk 1]
```diff
@@ -181,204 +148,246 @@ final class ActorMqttClientSession(settings: MqttSessionSettings)(implicit mat:
...
+                }
+                .filter(_.nonEmpty)
+                .log("client-commandFlow", _.iterator.decodeControlPacket(settings.maxPacketSize)) // we decode here so we can see the generated packet id
+                .withAttributes(ActorAttributes.logLevels(onFailure = Logging.DebugLevel))
+          }
+          .recover {
+            case e: TimeoutException =>
+              val f: CommandFlow[A] =
+                Flow.fromSinkAndSource(Sink.ignore, Source.empty[ByteString])
+              f
```
Comment (ennru) [inline mqtt-streaming/src/main/scala/akka/stream/alpakka/mqtt/streaming/scaladsl/MqttSession.scala]: This hack makes the shutdown test
https://github.com/akka/alpakka/blob/9a87c963dc83fae259a372b59918f70e1f57e979/mqtt-streaming/src/test/scala/docs/scaladsl/MqttSessionSpec.scala#L1256
behave as before. But the correct behaviour should be I failure when someone tries to use a session that is shut down already, right?
cc @huntc

**Comment by huntc** [comment]
My first impression is that this adds complexity purely to support binary compatibility. If we didn’t have to support both 2.5 and 2.6 then we wouldn’t need this hack, right?

**Comment by ennru** [comment]
I'm not sure Akka 2.5/2.6 can be done even with this. But to put the routers into the connectors looks better to me than lots of system-level actors.
I think about going Akka 2.6 only.

**Comment by huntc** [comment]
Routers are a session level abstraction, so they should be able to live as children where you have them then. However, I’d look them toward hiding the routers from the outside.

**Comment by ennru** [comment]
As I don't think this will solve the Akka 2.5/2.6 problem, I close this.

**Pull request closed without merging**